docs/DesignDocs/OpteeIntegration.md

```diff
@@ -0,0 +1,321 @@
+TA is assumed to be in the new format (and the name of that field is
+`depr_entry`); if it is not, it is presumed to be in the old format. The old
+format is referred to as "Legacy TA" and the new format as "Bootstrap TA".
+
+One may conclude from this that there is no formal method for a TA binary to
+express which version of the TA loader it was built for.
+
+### TA Signing
+
+The signing and optional encryption mechanism is described (rather briefly)
+in section 2.10.2 of the
+[OP-TEE documentation](https://optee.readthedocs.io/_/downloads/en/latest/pdf/)).
+The format is made quite clear by reading the
+[signature script](https://github.com/OP-TEE/optee_os/blob/3.6.0/scripts/sign.py) and
+[encryption script](https://github.com/OP-TEE/optee_os/blob/master/scripts/sign_encrypt.py).
+Note that TA encryption was added post-OP-TEE 3.6.0. As such, this feature
+is not yet supported by Open Enclave, which is based on OP-TEE 3.6.0.
+
+#### TA Loader & Signing
+
+The verification of a TA's signature and its decryption are not performed by
+the TA loader. Rather, they are performed by the REE TA store. The REE TA
+store takes care of implementing the support for signature validation and
+decryption. By the time the TA binary reaches the TA loader, it is in
+cleartext and the signature and encryption header has been discarded.
+
+#### Custom Store / Loader
+
+It would be possible to create a new TA store that implements Open Enclave
+style metadata support, though that would require an Open Enclave specific
+change to OP-TEE itself.  However, that would probably not be sufficient.
+For example, if we must standardize on how to represent the required stack
+size for an enclave, that would require a change in the TA loader as well
+unless we standardize on the representation already used by OP-TEE.
+Hence, a new TA store or modification of the current one as well as changes
+to the TA loader might be required.
+
+#### Convergence
+
+Issue #2041 tracks the fact that `oesign` is SGX-specific and that signing
+OP-TEE TAs is completely different. While one could rewrite OP-TEE's signature
+script in C, it would seem preferable to invoke it from oesign. The reason is
+that the script may change from release to release and is thus a moving
+target. If we were to converge on a single metadata format, then the problem
+would of course go away.
+
+With regards to backwards compatibility in the metadata, OP-TEE follows
+GlobalPlatform's specification. GlobalPlatform explicitly calls out backwards
+compatibility in its specs, so there should be no breaking changes regarding
+the properties (i.e., `ta_props`).
+
+Regarding the TA binary and TA headers, which are not specified by
+GlobalPlatform, there is no specific call-out for backwards compatibility.
+However, Linaro's behavior so far indicates that they do not wish for TAs
+built for old versions of OP-TEE to break when run on newer versions. The
```

Comment:
I would suggest rephrasing this sentence as:

However, OP-TEE continues to ensure that TAs built for old versions for OP-TEE don't break when run on newever versions.

Reply:
Done